src/neighbors.cc

```diff
@@ -1503,78 +1503,78 @@ static void
 dump_peers(StoreEntry *sentry, CachePeers *peers)
 {
     char ntoabuf[MAX_IPSTRLEN];
+    PackableStream os(*sentry);
     int i;
 
     if (!peers) {
-        storeAppendPrintf(sentry, "There are no neighbors installed.\n");
+        os << "There are no neighbors installed.\n";
         return;
     }
 
     for (const auto &peer: *peers) {
         const auto e = peer.get();
         assert(e->host != nullptr);
-        storeAppendPrintf(sentry, "\n%-11.11s: %s\n",
-                          neighborTypeStr(e),
-                          e->name);
-        storeAppendPrintf(sentry, "Host       : %s/%d/%d\n",
-                          e->host,
-                          e->http_port,
-                          e->icp.port);
-        storeAppendPrintf(sentry, "Flags      :");
+        os << "\n" << std::setw(11) << std::left <<
+            neighborTypeStr(e) << ": " << e->name << "\n";
+        os << "Host       : " << e->host << '/' << e->http_port << '/' <<
+            e->icp.port << "\n";
+        os << "Flags      :";
+        os.flush();
         dump_peer_options(sentry, e);
 
-        for (i = 0; i < e->n_addresses; ++i) {
-            storeAppendPrintf(sentry, "Address[%d] : %s\n", i,
-                              e->addresses[i].toStr(ntoabuf,MAX_IPSTRLEN) );
-        }
+        for (i = 0; i < e->n_addresses; ++i)
+            os << "Address[" << i << "] : " << e->addresses[i].toStr(ntoabuf, MAX_IPSTRLEN) << "\n";
```

Comment:
Won't this have the same problem of emitting an URL instead of the naked IP?
I have another draft PR aiming to change the awkward API to instead return a SBuf (as an additional call / wrapper, at least during the transition)

Reply:
> Won't this have the same problem of emitting an URL instead of the naked IP?

It is difficult for me to answer this loaded question in a meaningful way. FWIW, my change request does not contradict https://github.com/squid-cache/squid/pull/1730#discussion_r1519722148. If you see specific problems that make the requested switch a bad idea, let's discuss!

----

> I have another draft PR aiming to change the awkward API to instead return a SBuf (as an additional call / wrapper, at least during the transition)

That plan raises a few red flags for me. FWIW, I recommend limiting the corresponding initial changes to a couple of use cases to avoid rewriting a lot of code.